recipes/parallelio/meta.yaml

```diff
@@ -0,0 +1,74 @@
+{% set version = "2.5.8" %}
+{% set ver = version.replace(".", "_") %}
+{% set build = 0 %}
+
+# recipe-lint fails if mpi is undefined
+{% set mpi = mpi or 'mpich' %}
+{% if mpi == "mpich" %}
+  # prioritize mpich via build number
+  {% set build = build + 100 %}
+{% endif %}
+
+{% set mpi_prefix = "mpi_" + mpi %}
+
+package:
+  name: parallelio
+  version: {{ version }}
+
+source:
+  url: https://github.com/NCAR/ParallelIO/archive/refs/tags/pio{{ ver }}.tar.gz
+  sha256: f2584fb4310ff7da39d51efbe3f334efd0ac53ae2995e5fc157decccc0570a89
+
+build:
+  number: {{ build }}
+  skip: True  # [win]
+  # add build string so packages can depend on mpi variants dependencies:
+  # `PKG_NAME * mpi_mpich_*` for mpich
+  # `PKG_NAME * mpi_*` for any mpi
+  string: "{{ mpi_prefix }}_h{{ PKG_HASH }}_{{ build }}"
+  run_exports:
+    - {{ pin_subpackage('parallelio', max_pin='x.x.x') }} {{ mpi_prefix }}_*
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+    - {{ compiler('fortran') }}
+    - cmake
+    - make
+    - llvm-openmp  # [osx]
+    - libgomp      # [linux]
+  host:
+    - {{ mpi }}
+    # these need to be listed twice so conda-smithy build picks up the pins
+    - hdf5
+    - hdf5 * {{ mpi_prefix }}_*
+    - libnetcdf
+    - libnetcdf * {{ mpi_prefix }}_*
+    - netcdf-fortran
+    - netcdf-fortran * {{ mpi_prefix }}_*
```

Comment:
Check the logs. The link inspector thinks you are overdepending on netcdf-fortran and hdf5 because it can't find any dynamic links to these libraries. Are y'all doing static linking?

Reply:
@carterbox, you may be right that `hdf5` isn't needed.  It isn't explicitly referenced in the `cmake` command.

Regarding `netcdf-fortran`, this is clearly a dependency.  The problem could easily lie with that library rather than here.  It would appear the static and shared libraries are both being included in that package:
https://github.com/conda-forge/netcdf-fortran-feedstock/blob/main/recipe/build.sh#L32-L58
As a result, presumably the static library is being linked here instead of the dynamic?

I won't have time for awhile but I can look into splitting that package into only the dynamic library and a separate `netcdf-fortran-static` package.

Reply:
This is turning out to be more complicated than I hoped.  The c library has not yet been split into separate dynamic and static packages:
https://github.com/conda-forge/libnetcdf-feedstock/pull/135
That's presumably a prerequisite for splitting the fortran library, since it has the c library as a dependency.

Reply:
Looks like the static libs were removed after the latest build on this PR. I will restart the build to see if it builds successfully without static libs for netcdf-fortran.

https://github.com/conda-forge/netcdf-fortran-feedstock/pull/76

Reply:
[Logs are still showing no dynamic linkage to netcdf-fortran.
](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=598878&view=logs&j=6f142865-96c3-535c-b7ea-873d86b887bd&t=22b0682d-ab9e-55d7-9c79-49f3c3ba4823&l=1486) We should be able to detect a link to `libnetcdff.so.7` if it is actually being used.